text/0000-networkobject-parenting.md

```diff
@@ -0,0 +1,222 @@
+- Feature Name: `networkobject-parenting`
+- Start Date: 2021-05-26
+- RFC PR: [Unity-Technologies/com.unity.multiplayer.rfcs#0000](https://github.com/Unity-Technologies/com.unity.multiplayer.rfcs/pull/0000)
+- Issue: [Unity-Technologies/com.unity.multiplayer#0000](https://github.com/Unity-Technologies/com.unity.multiplayer/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes a way to reparent `NetworkObject`s to other `NetworkObject`s at runtime while networking.
+
+The approach we are presenting here is limited by current MLAPI architecture and its internal structure. We will touch on these limitations and reasons below.
+
+# Motivation
+[motivation]: #motivation
+
+We want to offer `NetworkObject` reparenting solution within MLAPI to help developers synchronizing `transform` parent-child relationships of `NetworkObject`s.
+
+It is often not easy to grok the parenting in networking, also your choice of networking framework would have its limitations you don't necessarily know up until trying to implement and cover the edge cases. Here, we want to provide a solution that plays nicely with current MLAPI limitations and we expect this "NetworkObject Parenting" concept to evolve and change over time when MLAPI matures.
+
+With this RFC implemented, we expect developers to reparent their `NetworkObject`s at runtime during networked gameplay and synchronize across the network (server to clients) out-of-the-box within certain limits and conditions.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Rules
+
+Let's list a few basic `NetworkObject` reparenting rules below.
+
+### Only A Server (or A Host) Can Reparent
+
+Similar to Ownership, only the server (or host, which is both a server and a client at the same time) can control reparenting of a `NetworkObject` in the network.
+
+Clients however, can send RPCs to server and execute a logic server-side that ultimately makes server to reparent a `NetworkObject`.
+
+### Only Reparenting Under A NetworkObject (Or To The Root) Is Valid
+
+A `NetworkObject` can only be reparented under another `NetworkObject` (`GameObject` with `NetworkObject` component attached). Only exception is moving a `NetworkObject` to the root of the scene hierarchy.
+
+This is simply due to the fact that MLAPI would not be able to identify & locate new parent on the remote-side if it was a non-`NetworkObject` parent. Again, except moving it to the root because we could identify no parent (root) scenario without `NetworkObject` identification or scene hierarchy traversal.
+
+### Only Reparenting During Networking Is Valid
+
+A `NetworkObject` can only be reparented while networking, in other terms you can only reparent while listening/running as a server.
```

Comment:
Can an unspawned `NetworkObject` on the server be reparented while the server is listening/running?

Reply:
nope.